framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3170,6 +3171,47 @@ NonlinearSystemBase::checkKernelCoverage(const std::set<SubdomainID> & mesh_subd
   _scalar_kernels.subdomainsCovered(input_subdomains, kernel_variables);
   _constraints.subdomainsCovered(input_subdomains, kernel_variables);
 
+  if (_fe_problem.haveFV())
+  {
+    std::vector<FVElementalKernel *> fv_elemental_kernels;
+    _fe_problem.theWarehouse()
+        .query()
+        .template condition<AttribSystem>("FVElementalKernel")
+        .queryInto(fv_elemental_kernels);
+
+    for (auto fv_kernel : fv_elemental_kernels)
+    {
+      if (fv_kernel->blockRestricted())
+        for (auto block_id : fv_kernel->blockIDs())
+          input_subdomains.insert(block_id);
+      else
+        global_kernels_exist = true;
+      kernel_variables.insert(fv_kernel->variable().name());
+
+      // Check for lagrange multiplier
+      if (dynamic_cast<FVScalarLagrangeMultiplier *>(fv_kernel))
+        kernel_variables.insert(
+            dynamic_cast<FVScalarLagrangeMultiplier *>(fv_kernel)->lambda_variable().name());
```

Comment:
I don't love this but it certainly is an easy way to do it. If we start to see more of this from other objects that contribute to multiple variable residuals, then I think we'll want to do this differently.